doc/certificate/CERTIFICATES_README.md

```diff
@@ -0,0 +1,572 @@
+# Certificate Generation Guide
+
+## Overview
+
+This directory contains a script to generate a complete PKI (Public Key Infrastructure) hierarchy for the DSP True Connector:
+
+```
+Root CA (self-signed)
+    ↓ signs
+Intermediate CA
+    ↓ signs
+Server Certificates (connector-a, connector-b)
+```
+
+## Files
+
+- `generate-certificates.cmd` - Main script to generate all certificates
+- `setup-truststore.cmd` - Legacy script (replaced by generate-certificates.cmd)
+
+## Quick Start
+
+### 1. Edit Configuration (Optional)
+
+Open `generate-certificates.cmd` and edit the configuration section at the top:
+
+```batch
+REM Subject Alternative Names (SAN) - Edit these lists as needed for each service
+REM Each server should only have the SANs it actually needs for security best practices
+set SAN_CONNECTOR_A=DNS:localhost,DNS:connector-a,IP:127.0.0.1
+set SAN_CONNECTOR_B=DNS:localhost,DNS:connector-b,IP:127.0.0.1
+set SAN_MINIO=DNS:localhost,DNS:minio,IP:127.0.0.1
+
+REM Root CA Distinguished Name
+set ROOT_DNAME=CN=DSP Root CA, OU=Security, O=DSP True Connector, L=Belgrade, ST=Serbia, C=RS
+
+REM Validity periods (in days)
+set ROOT_VALIDITY=3650        # 10 years
+set INTERMEDIATE_VALIDITY=1825 # 5 years
+set SERVER_VALIDITY=365        # 1 year
+
+REM Passwords (change these for production!)
+set ROOT_PASSWORD=password
+set INTERMEDIATE_PASSWORD=password
+set SERVER_PASSWORD=password
+set TRUSTSTORE_PASSWORD=password
+```
+
+**Note:** Each server certificate now has **specific SANs** for better security. The certificate for `connector-a` can only serve requests for `localhost`, `connector-a`, and `127.0.0.1`. This prevents certificate misuse and follows security best practices.
+
+### 2. Run the Script
+
+```cmd
+cd connector/src/main/resources
+generate-certificates.cmd
+```
+
+The script will:
+1. ✅ Generate Root CA (self-signed)
+2. ✅ Generate Intermediate CA (signed by Root CA)
+3. ✅ Generate server certificates for connector-a and connector-b (signed by Intermediate CA)
+4. ✅ Create truststore with Intermediate CA certificate
+5. ✅ Verify all certificates
+
+### 3. Generated Files
+
+After running the script, you'll have:
+
+| File | Description | Usage |
+|------|-------------|-------|
+| `dsp-root-ca.p12` | Root CA keystore | **Keep secure!** Used only for signing Intermediate CA |
+| `dsp-intermediate-ca.p12` | Intermediate CA keystore | **Keep secure!** Used for signing server certificates |
+| `connector-a.p12` | Server certificate for connector-a | Use in consumer application |
+| `connector-b.p12` | Server certificate for connector-b | Use in provider application |
+| `dsp-truststore.p12` | Truststore with Intermediate CA | Use for TLS validation in both applications |
```

Comment:
We should make separate truststores, each instance of the connector should have it's own. It would be similar to a production setup and with a clear separation we would not introduce possible mistakes as to when and which certificate is used.